test/cpp/end2end/xds/xds_fallback_end2end_test.cc

```diff
@@ -65,21 +62,26 @@ class XdsFallbackTest : public XdsEnd2endTest {
     XdsEnd2endTest::TearDown();
   }
 
-  std::string SetupServer(BalancerServerThread* balancer, size_t backend,
-                          int server_id) {
+  // Sets resources and returns target name
```

Comment:
I think the tests would be a bit more readable if we encoded the target name in the tests themselves and passed it in here, rather than constructing it here and returning it.  I'm thinking something like this:

```
void SetXdsResourcesForTarget(BalancerServerThread* balancer, size_t backend,
                              absl::string_vierw target = "") {
  // ...
  if (!target.empty()) {
    listener.set_name(std::string(target));
    cluster.set_name(absl::StrCat("cluster_", target));
    cluster.mutable_eds_cluster_config()->set_service_name(
        absl::StrCat("eds_", target));
    route_config.set_name(absl::StrCat("route_", target));
    route_config.mutable_virtual_hosts(0)
        ->mutable_routes(0)
        ->mutable_route()
        ->set_cluster(cluster.name());
  }
  // ...
}
```

Then in individual tests, something like:

```
constexpr absl::string_view kServerName2 = "server2.example.com";
SetXdsResourcesForTarget(balancer_.get(), 2, kServerName2);
SetXdsResourcesForTarget(fallback_balancer_.get(), 3, kServerName2);
```